templates/themes/overboards/theme.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+	require 'info.php';
+
+	/**
+	 * Generate the board's HTML and move it and its JavaScript in place, whence
+	 * it's served
+	 */
+	function overboards_build($action, $settings) {
+		global $config;
+
+		if ($action !== 'all' && $action !== 'post' && $action !== 'post-thread' &&
+			$action !== 'post-delete')
+		{
+			return;
+		}
+
+		if ($config['smart_build']) {
+			file_unlink($settings['uri'] . '/index.html');
+		} else {
+			require 'overboards.php';
+
+			$overboards = new overboards($overboards_config);
+
+			foreach ($overboards_config as &$overboard) {
+				if (!file_exists($overboard['uri'])) {
+					@mkdir($overboard['uri'], 0777) or error("Couldn't create {$overboard['uri']}. Check permissions.", true);
+				}
+				// Copy the generated board HTML to its place
+				file_write($overboard['uri'] . '/index.html', $overboards->build($overboard));
+				file_write($overboard['uri'] . '/overboard.js',
+					Element('themes/overboards/overboard.js', array()));
+			}
+		}
+	}
+
+	/**
+	 * Encapsulation of the theme's internals
+	 */
+	class overboards {
+		private $settings;
+//TODO review if appropriate uses of pass by ref (&$)
```

Comment:
In line 57 you are modifying the parameters to the class. This is potentially problematic, and it might not be obvious when, leading to hard to reproduce bugs.